api/v1/storagecluster_types.go

```diff
@@ -75,6 +83,9 @@ type StorageClusterSpec struct {
 	// OverprovisionControl specifies the allowed hard-limit PVs overprovisioning relative to
 	// the effective usable storage capacity.
 	OverprovisionControl []OverprovisionControlSpec `json:"overprovisionControl,omitempty"`
+
+	// ApiServers defines the servers deployments which needs to be enabled
+	ApiServers map[ApiServerKind]bool `json:"apiServers,omitempty"`
```

Comment:
This doesn't seem to be very well defined. Even the term API server is very loaded and does not convey the entities that need to be deployed. And I am not even sure what should go into this map. 

I think a better API will be to have a boolean flag indicating that we are going to allow external storage consumers and the fact that we deploy an API server will be hidden as an implementation detail. 

So something like:
```Spec.AllowRemoteStorageConsumers = True/False```

Reply:
done